master/internal/trial_workload_sequencer.go

```diff
@@ -183,8 +183,8 @@ func (s *trialWorkloadSequencer) WorkloadCompleted(
 	}
 	if msg.ExitedReason != nil {
 		s.exitingEarly = true
-		if *msg.ExitedReason == workload.UserCanceled {
-			s.userRequestedStop = true
+		if *msg.ExitedReason == workload.UserCanceled || *msg.ExitedReason == workload.InvalidHP {
```

Comment:
could reuse the function mentioned above here, too, to capture the intent of this explicitly.